nativeApp/src/lib/hooks/requestHook.ts

```diff
@@ -0,0 +1,21 @@
+import { useState, useEffect } from 'react';
```

Comment:
seems like it would eventually be worth moving the generic hooks from the dashboard out so they can be re-used here. not in this PR tho I guess.

Reply:
I think the way we abstracted hooks in the dashboard is a little confusing. I was going to play around and see if I can find a simpler way to manage this in this app